nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -58,6 +58,8 @@ class SimpleActionServer
     auto handle_cancel =
       [this](std::shared_ptr<rclcpp_action::ServerGoalHandle<ActionT>>)
       {
+        std::lock_guard<std::mutex> lock(update_mutex_);
```

Comment:
Why are you taking the mutex here? This function only prints a message and returns a constant. No variables are referenced.

Reply:
As I understand, one side effect of returning `ACCEPT` is that the goal handle will switch to a cancel request state. Which possibly lead to a race condition.